projects/cli-files/go-cat/cmd/root.go

```diff
@@ -0,0 +1,36 @@
+package cmd
+
+import (
+	"os"
+	"bufio"
+	"fmt"
+	"flag"
+)
+
+func Execute() (error) {
+	lineNum := flag.Bool("n", false, "Number the output lines, starting at 1")
+	flag.Parse()
+
+	args := flag.Args()
+	for _, fileName := range args {
+		file, err := os.Open(fileName)
+		if err != nil {
+			return err
+		}
+		scanner := bufio.NewScanner(file)
+		printFileContent(*lineNum, scanner)
+	}
+	return nil
+}
+
+func printFileContent(displayLineNum bool, scanner *bufio.Scanner) (int) {
+	lineNumber := 0
+	for scanner.Scan() {
+		lineNumber++
+		if displayLineNum {
```

Comment:
Nice separation of the "always happens" behaviour from the "only sometimes happens" behaviour to make clear what's different!